File: back-end/app/src/main/java/com/senescyt/app/model/Funciones.java
package com.senescyt.app.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Funciones")
public class Funciones implements Serializable {

    /**
     *
     */
    private static final long serialVersionUID = 1L;
    /**
     *
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "funId")
    private Long funId;

    @Column(name = "funNombre")
    private String funNombre;


    @Column(name = "funEstado")
    private int funEstado;
    @JsonIgnore
    @OneToMany(mappedBy = "funId")
    private List<Usuario> listaUsuarios;

}

File: front-end/src/app/modelo/funciones.ts
export class Funciones {
    funId: number;
    funNombre: string;
    funEstado:number;
    
  
    constructor(funId?: number, funNombre?: string,funEstado?:number) {
      (this.funId = funId || 0),
        (this.funNombre = funNombre || ''),
        (this.funEstado = funEstado || 0);
        
    }
  }